Spec/Auth.swift

```diff
@@ -1903,6 +1903,32 @@ class Auth : QuickSpec {
                         done()
                     }
                 }
+                
+                tokenParams.clientId = "testId"
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(tokenParams, options: authOptions) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        guard let tokenDetails = tokenDetails else {
+                            XCTFail("TokenDetails is nil"); done(); return
+                        }
+
+                        expect(tokenDetails.clientId).to(equal("testId"))
+                        done()
+                    }
+                }
+                
+                tokenParams.clientId = nil
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(tokenParams, options: authOptions) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        guard let tokenDetails = tokenDetails else {
+                            XCTFail("TokenDetails is nil"); done(); return
+                        }
+                        
+                        expect(tokenDetails.clientId).to(beNil())
```

Comment:
> Ok, so I need to rename API from authorise to authorize (z letter is a diff) entirely in my current 0.8 scope, got it.

Well it's a 0.9 change, so we can defer to later, but we need an issue so that we don't mark this as completed mistakenly and never come back to it.  Can you do that please?